src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -862,5 +864,11 @@ private static Complex Scale(Complex value, double factor)
         {
             return new Complex((double)value, 0.0);
         }
+
+        public static bool IsFinite(Complex value) => !Infinity.Equals(value);
```

Comment:
A complex or imaginary value is a finite number if each of its parts is a finite number (neither infinite nor NaN).

Reply:
This can probably be implemented using `=> `double.IsFinite(value.m_real)  && double.IsFinite(value.m_imaginary)`